pkg/storage/minio/checker.go

```diff
@@ -8,12 +8,16 @@ import (
 	opentracing "github.com/opentracing/opentracing-go"
 )
 
-func (v *storageImpl) Exists(ctx context.Context, module, version string) bool {
+func (v *storageImpl) Exists(ctx context.Context, module, version string) (bool, error) {
 	sp, ctx := opentracing.StartSpanFromContext(ctx, "storage.minio.Exists")
 	defer sp.Finish()
 	versionedPath := v.versionLocation(module, version)
 	modPath := fmt.Sprintf("%s/go.mod", versionedPath)
 	_, err := v.minioClient.StatObject(v.bucketName, modPath, minio.StatObjectOptions{})
 
-	return err == nil
+	if minio.ToErrorResponse(err).Code == "NoSuchKey" {
```

Comment:
I'll write it down as you have conflicts otherwise i wouldn't bother you with this. can you make a const out of this?